velox/functions/sparksql/tests/ArrayInsertTest.cpp

```diff
@@ -0,0 +1,260 @@
+  testInt<float_t>();
+  testInt<double_t>();
+}
+
+TEST_F(ArrayInsertTest, boolArrays) {
+  const auto arrays = makeArrayVectorFromJson<bool>(
+      {"[true]", "[true, false]", "[true, false, true]"});
+
+  auto expected = makeArrayVectorFromJson<bool>(
+      {"[true, true]", "[true, true, false]", "[true, true, false, true]"});
+  testExpression("array_insert(c0, 1, true, false)", {arrays}, expected);
+
+  expected = makeArrayVectorFromJson<bool>(
+      {"[null, true]", "[null, true, false]", "[null, true, false, true]"});
+  testExpression(
+      "array_insert(c0, 1, null::BOOLEAN, false)", {arrays}, expected);
+
+  expected = expected = makeArrayVectorFromJson<bool>(
+      {"[true, true]", "[true, true, false]", "[true, false, true, true]"});
+  testExpression("array_insert(c0, -2, true, false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, strArrays) {
+  using S = StringView;
+
+  auto arrays = makeNullableArrayVector<StringView>({
+      {S("a"), std::nullopt, S("b")},
+      {S("a"), S("b"), S("a"), S("a")},
+      {std::nullopt, S("b"), std::nullopt},
+      {S("purple is an elegant color")},
+  });
+
+  auto expected = makeNullableArrayVector<StringView>({
+      {S("X"), S("a"), std::nullopt, S("b")},
+      {S("X"), S("a"), S("b"), S("a"), S("a")},
+      {S("X"), std::nullopt, S("b"), std::nullopt},
+      {S("X"), S("purple is an elegant color")},
+  });
+  testExpression("array_insert(c0, 1, 'X', false)", {arrays}, expected);
+
+  expected = makeNullableArrayVector<StringView>({
+      {std::nullopt, S("a"), std::nullopt, S("b")},
+      {std::nullopt, S("a"), S("b"), S("a"), S("a")},
+      {std::nullopt, std::nullopt, S("b"), std::nullopt},
+      {std::nullopt, S("purple is an elegant color")},
+  });
+  testExpression(
+      "array_insert(c0, 1, null::STRING, false)", {arrays}, expected);
+
+  expected = makeNullableArrayVector<StringView>({
+      {S("a"), std::nullopt, S("X"), S("b")},
+      {S("a"), S("b"), S("a"), S("X"), S("a")},
+      {std::nullopt, S("b"), S("X"), std::nullopt},
+      {S("X"), S("purple is an elegant color")},
+  });
+  testExpression("array_insert(c0, -2, 'X', false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, varbinary) {
+  auto arrays = makeNullableArrayVector<StringView>(
+      {{"a"_sv, "b"_sv, "c"_sv}}, ARRAY(VARBINARY()));
+
+  auto expected = makeNullableArrayVector<StringView>(
+      {{"X"_sv, "a"_sv, "b"_sv, "c"_sv}}, ARRAY(VARBINARY()));
+  testExpression(
+      "array_insert(c0, 1, 'X'::VARBINARY, false)", {arrays}, expected);
+
+  expected = makeNullableArrayVector<StringView>(
+      {{std::nullopt, "a"_sv, "b"_sv, "c"_sv}}, ARRAY(VARBINARY()));
+  testExpression(
+      "array_insert(c0, 1, null::VARBINARY, false)", {arrays}, expected);
+
+  expected = makeNullableArrayVector<StringView>(
+      {{"a"_sv, "b"_sv, "X"_sv, "c"_sv}}, ARRAY(VARBINARY()));
+  testExpression(
+      "array_insert(c0, -2, 'X'::VARBINARY, false)", {arrays}, expected);
+}
+
```

Comment:
Do we support complex types? Perhaps also add some tests.

Reply:
Sure, added tests cases for nested arrays and array of maps and structs. Please take a look.